tools/filterdiff.py

```diff
@@ -0,0 +1,150 @@
+# /usr/bin/env python3
+import sys
+import os
+import argparse
+import tempfile
+import json
+import subprocess
+from typing import List
+
+
+def filter_fn(imi, unwanted_tmpfiled_d: List):
```

Comment:
As a matter of preference, I would wrap all these filtering functions in a class as classmethods, instead of nesting them inside another function.

Reply:
Good point, I'll do that!